integration-cli/docker_utils.go

```diff
@@ -1567,3 +1567,23 @@ func sumAsIntegers(vals ...interface{}) interface{} {
 	}
 	return s
 }
+
+// pollFunc is used to periodically execute a check function, it
+// returns error after timeout.
+func pollFunc(f func() error, timeout time.Duration) error {
```

Comment:
Starting daemon waits for the http API to become active, this doesn't happen before swarm control API is active(unless you get timeout) and control api should wait for leader to be elected.